actions/inventory.go

```diff
@@ -119,12 +119,39 @@ func Collect(ctx context.Context, device *common.Device, collectors *Collectors,
 		return errors.Wrap(err, "error retrieving TPM inventory")
 	}
 
+	if dynamic {
+		// Update StorageControllerCollectors
+		for _, sc := range device.StorageControllers {
+			switch common.FormatVendorName(sc.Vendor) {
+			case common.VendorMarvell:
+				collectors.StorageControllers = utils.NewMvcliCmd(trace)
+				// TODO(splaspood) We need to handle multiple collectors here
+			}
+		}
+	}
+
 	// Collect StorageController info
```

Comment:
Could these dynamic bits be moved into the `StorageController()`, `Drive()` if those methods accepted the `dynamic` parameter?